### DISCUSSION TITLE
What is "performance" method in regards to jspsych-instructions plugin?

### DISCUSSION THREAD

**Discussion by choppr** [jspsych/jsPsych#947] [Q&A] [answered]
I get this console log: "Invalid RT method specified in getKeyboardResponse. Defaulting to "performance" method."
and I don't know why. It skips the ability to click next and immediately moves to the next timeline

**Comment by vijaymarupudi** [comment, marked as answer]
Can you post the code you are using for that trial? The performance method is a high-resolution method of measuring time in JavaScript.

**Reply by choppr** [reply]
``` 
   var intro = {
        type: 'instructions',
        pages: 
          ['<div style="font-size:20px;"> '+
          '<br><br><b>Welcome to the experiment!</b> '+
          // '<br><br><br>Continue the experiment remaining in fullscreen. '+
          '<br><br>Please complete the experiment on your own in one sitting, you will be given two optional breaks in between. '+
          '<br> Make sure you are in a quiet room with minimal distraction possible, including no music and other tasks at hand. '+
          '<br><br>Before we start, you will be given a chance to familiarize yourself with the experiment. '+
          '<br><br><i>Click ‘Next’ to continue.</i><br><br> </div>'],
        allow_backward: false,
        button_label_next: "Next",
        show_clickable_nav: true
      }
```

**Reply by vijaymarupudi** [reply]
Hmm, that looks fine to me, maybe the bug isn't where you think it is? Are you using the latest version of jsPsych? If so, could you post a screenshot of the error (after expanding it in the console).

**Reply by choppr** [reply]
I'm using Cognition.run with JsPsych library 6.1.0.

Here is the screenshot of my console.
![image](https://user-images.githubusercontent.com/38826941/89130837-28677e00-d4bd-11ea-8a13-a0cd5ce930c0.png)

I looked into the jspsych-instructions plugin and it looks like the value 'date' for the variable rt_method isn't supported anymore in jspsych

```
    var after_response = function(info) {

      // have to reinitialize this instead of letting it persist to prevent accidental skips of pages by holding down keys too long
      keyboard_listener = jsPsych.pluginAPI.getKeyboardResponse({
        callback_function: after_response,
        valid_responses: [trial.key_forward, trial.key_backward],
        rt_method: 'date',
        persist: false,
        allow_held_key: false
      });
```

**Reply by choppr** [reply]
Sometimes this console log error will appear many seconds after the instruction timeline finishes

**Reply by choppr** [reply]
After further inspection it looks like the jsPsych library being used on Cognition.run is different from the latest jsPsych library. I will raise an issue with Cognition.run

**Reply by vijaymarupudi** [reply]
Ah, I do not use cognition.run. Replicating using a local version can help!
